### PR TITLE
Implement "refine" substatement to "uses"

### PR DESCRIPTION
**Pull request by mzagozen**
After the "uses" statement is expanded, we also apply the refinements listed as substatements to "uses" to the target nodes in place.

### CODE REVIEW

File: gen/src/rfcgen.act [hunk 1]
```diff
@@ -845,6 +846,27 @@ snode_methods = {
         raise ValueError("Cannot compile 'uses'")
 
 """,
+        "expand_refines": '''    mut def expand_refines(self, target_base: SchemaNode):
+        """Expand refinements in the uses statement to the tqarget nodes
```
Comment (plajjan) [inline gen/src/rfcgen.act:850]:
```suggestion
        """Expand refinements in the uses statement to the target nodes
```

**Comment by plajjan** [comment]
I think this is sound. I will admit that it's not so easy for me to think about all the cases as when I wrote some of the expansion stuff initially. Perhaps we should have more complex test cases to discover more corner cases but I can't come up with anything immediately that would break.

**Comment by mzagozen** [comment]
Right, I don't know for sure either, but I think:
- default and mandatory are mutually exclusive on a node
- a node in an oper subtree may not be refined to config=true

**Comment by plajjan** [comment]
@mzagozen right, so some extra checks are nice but so far in acton-yang we have targetted compiling correct YANG models, not validating everything. Over time I think we should do more validation and be more error tolerant, but it is something we can grow into, we don't have to do it all in a day. You choose how much testing and extra checks you want to add now. I am fine merging as-is.
